### DISCUSSION TITLE
State Management with MAUI (MVU)

### DISCUSSION THREAD

**Discussion by ske66** [dotnet/maui#304] [General]
I'm super excited to start working with MAUI, I'll be trying it out the second the preview becomes available! One question I did have, what does the state management system look like? Will it be modelled off of existing global state management tools like Redux, or are you implementing your own way to process immutable data?

**Comment by jdebache** [comment]
I have the same question. I tried using Fluxor (https://github.com/mrpmorris/Fluxor), which works great in Blazor, but had no luck with it in MAUI so far.

**Comment by MhAllan** [comment]
You can use IServiceCollection to resolve dependencies, and Bindable/Notificiation properties and events to communicate changes. I don't think you need a framework for state management.

**Reply by jdebache** [reply]
You are right, there are other ways. But the Redux framework is easy to reason about, clean and testable. That is why it is preferred by many over hand rolled solutions such as the one you are suggesting. For this reason, I would be really happy if such a framework could support MAUI.
